src/coreclr/src/pal/src/synchobj/semaphore.cpp

```diff
@@ -238,6 +206,38 @@ CreateSemaphoreW(
     return hSemaphore;
 }
 
+/*++
```

Comment:
What's the reason for moving this around in the file?

Reply:
It was either forward declaration or moving the dependent function down.